workbench-core/base/src/utilities/queryParamsFiltersUtils.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { AttributeValue } from '@aws-sdk/client-dynamodb';
+import Boom from '@hapi/boom';
+import { QueryParams } from '../aws/helpers/dynamoDB/dynamoDBService';
+import { QueryParameterFilter } from '../interfaces/queryParameterFilter';
+
+/************************************************************
+ * Validates only one sorting or filter in list requests
+ * @param filter - filter request object
+ * @param sort - sorting request object
+ ************************************************************/
+export function validateSingleSortAndFilter(filter?: {}, sort?: {}): void {
+  if (filter && sort) {
+    throw Boom.badRequest('Cannot apply a filter and sort at the same time');
+  }
+  const filterAttributesLength = filter ? Object.keys(filter).length : 0;
+  // Check that at most one filter is defined because we not support more than one filter
+  if (filterAttributesLength > 1) {
+    throw Boom.badRequest('Cannot apply more than one filter.');
+  }
+  // Check that at most one sort attribute is defined because we not support sorting by more than one attribute
+  if (sort && Object.keys(sort).length > 1) {
+    throw Boom.badRequest('Cannot sort by more than one attribute.');
+  }
+}
+
+/************************************************************
+ * Parses Object with format \{ field: \{ operator: value \} \} into QueryParam object consumed by dynamoDB
+ *
+ * @param filter - filter request object field
+ * @param sortKey - field name in dynamoDB
+ * @param gsi - gsi to use for filter
+ *
+ * @returns QueryParams object
+ ************************************************************/
+export function parseQueryParamFilter<T>(
+  filter: QueryParameterFilter<T> | undefined,
+  sortKey: string,
+  gsi: string
+): QueryParams | undefined {
+  if (!filter || Object.keys(filter).length === 0 || Object.keys(filter).length > 1) {
+    return undefined;
```

Comment:
Would it be worth throwing an error here to indicate what went wrong with the parsing?